tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs

```diff
@@ -744,5 +784,44 @@ public void TestJoin()
             Assert.Equal(join["Int_right"][2], right["Int"][2]);
             VerifyJoin(join, left, right, JoinAlgorithm.Inner);
         }
+
+        [Fact]
+        public void TestGroupBy()
+        {
+            DataFrame df = MakeDataFrameWithNumericAndBoolColumns(10);
+            DataFrame count = df.GroupBy("Bool").Count();
+            Assert.Equal(2, count.RowCount);
+            Assert.Equal((long)5, count["Int"][0]);
+
+            DataFrame first = df.GroupBy("Bool").First();
+            Assert.Equal(2, first.RowCount);
+
+            DataFrame head = df.GroupBy("Bool").Head(3);
+            Assert.Equal(6, head.RowCount);
+
+            DataFrame tail = df.GroupBy("Bool").Tail(3);
+            Assert.Equal(6, tail.RowCount);
+
+            DataFrame max = df.GroupBy("Bool").Max();
+            Assert.Equal(2, max.RowCount);
+            Assert.Equal(8, max["Int"][0]);
+            Assert.Equal(9, max["Int"][1]);
+
+            DataFrame min = df.GroupBy("Bool").Min();
+            Assert.Equal(2, min.RowCount);
+            Assert.Equal(0, min["Int"][0]);
+            Assert.Equal(0, min["Int"][1]); // Because of the null value
```

Comment:
I'm not sure I get this comment. Can you elaborate more?

Also, if the behavior is surprising to warrant a comment, is there a bug here that should be addressed to make the behavior understandable?

Reply:
I filed this https://github.com/dotnet/corefxlab/issues/2700 as a follow up.

Consider a column of ints like [0, 1, 0, 1, null]. At the moment, this forms the groups : [0] and [1] instead of [0], [1] and [null]. `null` is considered equivalent to `default`. This is because the Dictionary used internally for grouping uses the following for it's key:
`dictionary.Add(column[rowIndex] ?? default, ...`

Once I fix https://github.com/dotnet/corefxlab/issues/2700, it should becomes more intuitive.